requests/models.py

```diff
@@ -373,7 +374,7 @@ def prepare_url(self, url, params):
         # Don't do any URL preparation for non-HTTP schemes like `mailto`,
         # `data` etc to work around exceptions from `url_parse`, which
         # handles RFC 3986 only.
-        if ':' in url and not url.lower().startswith('http'):
+        if SCHEME_RE.match(url) and not url.lower().startswith('http'):
```

Comment:
We parse the URL a few lines down. Why can't we parse it early and check if `scheme` is in `{"http", "https"}`? That will be the best test rather than trying to use a regular expression that isn't publicly documented as part of an API we can rely on